x-pack/plugins/observability_solution/observability_ai_assistant/docs/openapi/api.yaml

```diff
@@ -0,0 +1,126 @@
+openapi: 3.0.0
+info:
+  title: Elastic AI Assistant for Observability
+  version: 1.0.0
+paths:
+  "/api/observability_ai_assistant/chat/complete":
```

Comment:
Nice! Not sure it's in this PR code, what generated this spec?

Reply:
it's here: https://github.com/elastic/kibana/pull/183151/files#diff-012b20f075e09f40a1839b759082ffea0122aeb79d6b5fd5e642f9abee2435e2, from there it's mostly JSON to YML if needed which is fairly trivial

Reply:
we can do it automatically if we have an API from core, but I'm not sure if we can get to full coverage. custom io-ts types are ignored for instance.

Reply:
Oh sorry, my q was not clear: I meant the entire OAS document. The schema parts were generated, but the `openapi`, `info` parts were manually added by you?

Reply:
A random Q: I was checking [this code](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability_ai_assistant/server/routes/chat/route.ts#L291,L325) and I didn't see `access: public` parameter defined there, how is this API public?

Reply:
It starts with /api, and is assumed to be public unless specified otherwise (the inverse of that is /internal)

Reply:
That was my assumption before, but when I checked the infra plugin APIs, all of them start with /api but none of them are supposed to be public, and it seems whether they are public or not is controlled by `access` parameter which defaults to `internal` [here](https://github.com/elastic/kibana/blob/main/packages/core/http/core-http-server/src/router/route.ts#L144).

What is the implication of an API being public?

Reply:
Here is how APM plugin sets this field: https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/routes/apm_routes/register_apm_server_routes.ts#L232

Reply:
Yes this is also the case for the Obs AI Assistant routes which use @kbn/server-route-repository: https://github.com/elastic/kibana/blob/e66098a0a40882f0c511bf3dff4947e398f35131/packages/kbn-server-route-repository/src/register_routes.ts#L144. I thought Core did this too but I may have mis-interpreted a Slack thread.

Reply:
I will dm you the link, but for the readers here, the _convention_ is to use /api vs /internal for public vs internal routes, but I am not sure if that also means that `access` is set to one or the other based on the path name, from the docs it looks like it's not, but would need @jloleysens to clarify.

Reply:
This does not happen automatically. We will default to `access: internal` if not set in the router configuration. From 9.0 we will by default enable (current default is disable) a mechanism to require a special header to access internal endpoints ([issue](https://github.com/elastic/dev/issues/2669#issuecomment-2283993743)). So it is good you are setting `access` explicitly in the route repository.